test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -149,6 +150,29 @@ func (g *Grabber) GrabFromScheduler() (SchedulerMetrics, error) {
 	if g.kubeScheduler == "" {
 		return SchedulerMetrics{}, fmt.Errorf("kube-scheduler pod is not registered. Skipping Scheduler's metrics gathering")
 	}
+
+	var err error
+	podName := g.kubeScheduler
+	g.waitForSchedulerReadyOnce.Do(func() {
+		if readyErr := e2epod.WaitForPodsReady(g.client, metav1.NamespaceSystem, podName, 0); readyErr != nil {
```

Comment:
You shouldn't rely on pods being registered with the API Server if possible.  I believe the other fix you referenced waited until it could fetch metrics instead.

Reply:
Oh, looks like you do that below.  In that case, I don't think you need this initial pod ready check

Reply:
Thanks for the review.

I removed the `waiting for pod ready` here.
Updated.